script/zip-comp-logs

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+A script to zip the logs produced by a series of competition matches.
+"""
+
+import re
+import shutil
+import argparse
+import datetime
+import contextlib
+from typing import Union, Iterator
+from pathlib import Path
+from zipfile import ZipFile
+
+
+def zip_folder(folder: Path, args: argparse.Namespace) -> str:
+    tla = folder.name
+    zip_name = f'team-{tla}'
+
+    if args.suffix:
+        zip_name += f'-{args.suffix}'
+    if args.include_date:
+        zip_name += f'-{datetime.date.today()}'
+
+    shutil.make_archive(
+        f'{args.output_dir / zip_name}',
+        'zip',
+        root_dir=folder,
+        base_dir='.',
+    )
+    return f'{zip_name}.zip'
+
+
+def is_team_logs_folder(folder: Path) -> bool:
+    # Options for detecting which folders are logs:
+    # 1. everthing except recordings, matches
+    # 2. folders w/ matching zip file names
+    # 3. regex of folder name
+    # 4. regex of folder contents names
+
+    # Using option 3
+    # folder names are three uppercase letters and an optional number
+    if re.match(r'[A-Z]{3}[0-9]*$', folder.name):
+        return True
+
+    return False
+
+
+@contextlib.contextmanager
+def optional_zip(filename: Path, create_archive: bool) -> Iterator[Union[None, ZipFile]]:
+    if create_archive:
+        with ZipFile(filename, 'w') as zipfile:
+            yield zipfile
+    else:
+        yield None
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'archives_dir',
+        help=(
+            "The directory containing folders of the teams' robot logs, "
+            "named for the teams' TLAs. This directory also contains the "
+            "teams' robot code, as Zip archives and the match recordings."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        'output_dir',
+        help=(
+            "The directory to place the resulting log Zip archives. "
+            "This should be an empty directory."
+        ),
+        type=Path,
+    )
+    parser.add_argument(
+        '--suffix',
+        help=(
+            "Adds a suffix to the Zip archive name. "
+            "Archive names follow the format: team-<TLA>-<suffix>-<date>.zip"
+        ),
+        default="",
+    )
+    parser.add_argument(
+        '--no-date',
+        help=(
+            "Remove the date from the archive name."
+        ),
+        dest='include_date',
+        action='store_false',
+    )
+    parser.add_argument(
+        '--combined',
```

Comment:
Maybe this should be `--with-combined`? Else it sort of implies it's _only_ creating the combined archive